featurebyte/models/credential.py

```diff
@@ -73,6 +73,15 @@ def encrypt(self) -> None:
         for field in self.__fields__.values():
             if field.type_ == StrictStr:
                 setattr(self, field.name, encrypt_value(getattr(self, field.name)))
+            elif field.type_ == str:
+                field_value = getattr(self, field.name)
+                if isinstance(field_value, dict):
```

Comment:
Can this still evaluate to true given the `field.type_ == str` condition above?

Reply:
It's strange for me too seems the pydantic `field.type_` attribute is str even though the field is declared as `Dict[str, str]`

Reply:
I see. Maybe worth adding a comment. Btw curious why the GCSStorageCredential structure has a nested dictionary structure, as opposed to a flattened one like S3StorageCredential?

Reply:
The GCS service account is a json dict (not nested). It's usually provided as a file path but since we need to store this in the db that won't work. Another option is to change the parameter into a json string. The client that we use accepts either a file path or a python dictionary